interfaces/cython/cantera/base.pyx

```diff
@@ -232,6 +232,21 @@ cdef class _SolutionBase:
         def __get__(self):
             return anymap_to_dict(self.base.parameters(True))
 
+    def update_user_data(self, data):
+        """
+        Add the contents of the provided `dict` as additional fields when generating
+        YAML phase definition files with `write_yaml` or in the data returned by
+        `input_data`.
+        """
```

Comment:
We should document what happens when duplicate keys are present. Presumably, the key is updated with the new value, but it'd be good to be explicit about explaining that.

Reply:
Also, whether this is a recursive update, or if it will overwrite sub-fields